plugins/generator-1.21.4/neoforge-1.21.4/templates/item/overrides_model.java.ftl

```diff
@@ -0,0 +1,145 @@
+ # exception.
+-->
+
+<#-- @formatter:off -->
+package ${package}.client.renderer.item;
+
+<#compress>
+@EventBusSubscriber(bus = EventBusSubscriber.Bus.MOD, value = Dist.CLIENT) public class LegacyOverrideSelectItemModel implements ItemModel {
+
+    private final ModelOverride[] overrides;
+    private final ItemModel[] models;
+    private final ItemModel fallback;
+
+    private LegacyOverrideSelectItemModel(ModelOverride[] overrides, ItemModel[] models, ItemModel fallback) {
+        this.overrides = overrides;
+        this.models = models;
+        this.fallback = fallback;
+    }
+
+    @Override public void update(ItemStackRenderState renderState, ItemStack itemStack, ItemModelResolver modelResolver,
+            ItemDisplayContext displayContext, @Nullable ClientLevel level, @Nullable LivingEntity entity, int seed) {
+        ItemModel model = fallback;
+        for (int i = 0; i < overrides.length; i++) {
+            if (overrides[i].test(itemStack, level, entity, seed, displayContext))
+                model = models[i];
+        }
+        model.update(renderState, itemStack, modelResolver, displayContext, level, entity, seed);
+    }
+
+    @OnlyIn(Dist.CLIENT) public record FloatEntry(RangeSelectItemModelProperty property, float value) implements PredicateEntry {
+        public static final Codec<LegacyOverrideSelectItemModel.FloatEntry> CODEC = RecordCodecBuilder.create(
+            instance -> instance.group(
+                RangeSelectItemModelProperties.MAP_CODEC.fieldOf("property").forGetter(LegacyOverrideSelectItemModel.FloatEntry::property),
+                Codec.FLOAT.fieldOf("value").forGetter(LegacyOverrideSelectItemModel.FloatEntry::value)
+            ).apply(instance, LegacyOverrideSelectItemModel.FloatEntry::new)
+        );
+
+        @Override public boolean test(ItemStack itemStack, @Nullable ClientLevel level, @Nullable LivingEntity entity, int seed, ItemDisplayContext displayContext) {
+            return property.get(itemStack, level, entity, seed) >= value;
+        }
+    }
+
+    @OnlyIn(Dist.CLIENT) public record BoolEntry(ConditionalItemModelProperty property, boolean value) implements PredicateEntry {
+        public static final Codec<LegacyOverrideSelectItemModel.BoolEntry> CODEC = RecordCodecBuilder.create(
+            instance -> instance.group(
+                ConditionalItemModelProperties.MAP_CODEC.fieldOf("property").forGetter(LegacyOverrideSelectItemModel.BoolEntry::property),
+                Codec.BOOL.fieldOf("value").forGetter(LegacyOverrideSelectItemModel.BoolEntry::value)
+            ).apply(instance, LegacyOverrideSelectItemModel.BoolEntry::new)
+        );
+
+        @Override public boolean test(ItemStack itemStack, @Nullable ClientLevel level, @Nullable LivingEntity entity, int seed, ItemDisplayContext displayContext) {
+            return property.get(itemStack, level, entity, seed, displayContext) || !value;

```

Comment:
Are you sure this works the right way. If the value is "false", this predicate will pass in all cases, no matter the value

Reply:
If we translate this expression to 1.21.1 format, values of `true` and `false` would become `1.0` and `0.0`:
```java
            return property.get(itemStack, level, entity, seed, displayContext) == 1.0 || value == 0.0;
```